plugin/forward/setup.go

```diff
@@ -211,6 +213,19 @@ func parseBlock(c *caddy.Controller, f *Forward) error {
 		default:
 			return c.Errf("unknown policy '%s'", x)
 		}
+	case "max_concurrent":
+		if !c.NextArg() {
+			return c.ArgErr()
+		}
+		n, err := strconv.Atoi(c.Val())
+		if err != nil {
+			return err
+		}
+		if n < 0 {
+			return fmt.Errorf("max_concurrent can't be negative: %d", n)
+		}
+		ErrLimitExceeded = errors.New("concurrent queries exceeded maximum " + c.Val())
```

Comment:
Here may be a race condition. When coredns is reloaded (e.g. with SIGUSR1) the old instance of forward plugin is still handling traffic and may access `ErrLimitExceeded`, and the new instance of forward plugin can update `ErrLimitExceeded` at the same time